jobhub.go

```diff
@@ -0,0 +1,121 @@
+package jobhub
+
+import (
+	//"github.com/cenkalti/backoff"
+	//"github.com/orian/utils/common"
+	"github.com/sirupsen/logrus"
+	"os/exec"
+	"syscall"
+	"time"
+)
+
+/*
+type PipelineManager interface {
+	New(string) Pipeline
+	AddJob(Job) Job
+	AddJobDependency(...Job)
+	Run()
+}
+*/
+
+var nextPipelineID int = 1
+
+type Pipeline struct {
+	Name          string
+	Log           logrus.FieldLogger
+	id            int
+	nextJobID     int
+	jobContainer  []Job
+	jobDependency []Job
+}
+
+type Job struct {
+	Name       string
+	Path       string
+	pipelineID int
+	id         int
+}
+
+type exitStatus struct {
+	runtime time.Duration
+	status  syscall.WaitStatus
+}
+
+func (p *Pipeline) runJob(job Job) (*exitStatus, error) {
+	_, err := exec.LookPath(job.Path)
```

Comment:
exec.LookPath searches for execs in `$PATH` it may be useful, or may be not. It's not the same as checking if a file exists and is executable.

You need to use `os.Stat` to just check if file exists.